src/ferc_xbrl_extractor/datapackage.py

```diff
@@ -346,44 +351,34 @@ def construct_dataframe(self, instance: Instance) -> pd.DataFrame:
         Args:
             instance: Parsed XBRL instance used to construct dataframe.
         """
-        # Filter contexts to only those relevant to table
-        contexts = instance.get_facts(self.instant, self.axes)
-
-        # Get the maximum number of rows that could be in table and allocate space
-        max_len = len(contexts)
-
-        # Allocate space for columns
-        df = {
-            field.name: pd.Series([pd.NA] * max_len, dtype=FIELD_TO_PANDAS[field.type_])
-            for field in self.schema.fields
-        }
-
         # Loop through contexts and get facts in each context
         # Each context corresponds to one unique row
-        for i, (context, facts) in enumerate(contexts.items()):
-            if self.instant != context.period.instant:
-                continue
-
-            # Construct dictionary to represent row which corresponds to current context
-            row = {fact.name: fact.value for fact in facts if fact.name in df}
-
-            # If row is empty skip
-            if row:
-                # Use context to create primary key for row and add to dictionary
-                row.update(context.as_primary_key(instance.filing_name))
-
-                # Loop through each field in row and add to appropriate column
-                for key, val in row.items():
-                    # Try to parse value from XBRL if not possible it will remain NA
-                    try:
-                        df[key][i] = self.columns[key](val)
-                    except ValueError:
-                        self.logger.warning(
-                            f"Could not parse {val} as {self.columns[key]} in column {key}"
-                        )
-
-        # Create dataframe and drop empty rows
-        return pd.DataFrame(df).dropna(how="all")
+        df = {}
```

Comment:
This whole "make a dictionary, fill the columns with a nested for-loop, and then turn that dictionary into a dataframe" thing seemed like a lot of steps, for the goal of "make a dataframe with a bunch of facts."

Reply:
I do wonder, though, if we want to keep this dataframe stacked/narrow. The unstacked version can get pretty sparse for a lot of forms where each context has one fact for a different concept.

The stacked version would also potentially be much more robust to future taxonomy changes - so long as the primary key is the same / only added to, the concept names can fluctuate and we can just store them all in the same table!

I think that change would definitely be out of scope of this PR.